### PR TITLE
Daemonset helper function that returns pods that DaemonSet controller would start on the given node.

### PR DESCRIPTION
**Pull request by mwielgus**
ref: #43

### CODE REVIEW

File: cluster-autoscaler/utils/daemonset/daemonset.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+			glog.Errorf("Failed with %v", err)
+		}
+	}
+	return result
+}
+
+func newPod(ds *extensionsv1.DaemonSet, nodeName string) *apiv1.Pod {
+	newPod := &apiv1.Pod{Spec: ds.Spec.Template.Spec, ObjectMeta: ds.Spec.Template.ObjectMeta}
+	newPod.Namespace = ds.Namespace
+	newPod.Name = ds.Name + "-pod"
```
Comment (MaciekPytel) [inline cluster-autoscaler/utils/daemonset/daemonset.go]: Can we get make this more unique (by adding node name or random component)? It probably won't break anything immediately, but the names should be unique within namespace and I'd rather avoid the risk of major wtf when debugging something unrelated in a year.

File: cluster-autoscaler/utils/daemonset/daemonset.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+// GetDaemonSetPodsForNode returns daemonset nodes for the given pod.
+func GetDaemonSetPodsForNode(nodeInfo *schedulercache.NodeInfo, daemonsets []*extensionsv1.DaemonSet, predicateChecker *simulator.PredicateChecker) []*apiv1.Pod {
+	result := make([]*apiv1.Pod, 0)
+	for _, ds := range daemonsets {
+		pod := newPod(ds, nodeInfo.Node().Name)
+		if err := predicateChecker.CheckPredicates(pod, nodeInfo); err == nil {
+			result = append(result, pod)
+		} else {
+			glog.Errorf("Failed with %v", err)
```
Comment (MaciekPytel) [inline cluster-autoscaler/utils/daemonset/daemonset.go]: This log message is very non-specific, can you make it more descriptive and include the name of daemonset and node?

Comment (MaciekPytel) [inline cluster-autoscaler/utils/daemonset/daemonset.go]: Also do we want to log errors just because a daemonset uses nodeSelector??

Comment (mwielgus) [inline cluster-autoscaler/utils/daemonset/daemonset.go]: Artifact from debugging. Removed.

File: cluster-autoscaler/utils/daemonset/daemonset.go (hunk 1 above)
Comment (mwielgus) [inline cluster-autoscaler/utils/daemonset/daemonset.go]: Done.

**Pull request merged**